velox/functions/prestosql/tests/ArrayIntersectTest.cpp

```diff
@@ -278,3 +311,87 @@ TEST_F(ArrayIntersectTest, wrongTypes) {
 
   EXPECT_NO_THROW(testExpr(expected, "array_intersect(C0, C0)", {array1}));
 }
+
+TEST_F(ArrayIntersectTest, arrayOfBigintArrays) {
+  auto baseVector = makeNullableArrayVector<int64_t>(
+      {// 1st row: two sets with some overlap
+       {1, 2, 3},
+       {2, 3},
+       // 2nd row: three sets with some overlap and duplicates
+       {3, 3, 5, 6},
+       {4, 4, 5, 5, 6},
+       {6, 6, 7},
+       // 3rd row: two sets with no overlap but duplicates
+       {5, 5},
+       {6, 6},
+       // 4th row: three sets with null and non-null overlap
+       {1, 2, std::nullopt, std::nullopt},
+       {2, std::nullopt},
+       {2, 3, std::nullopt},
+       // 5th row: two sets with null overlap and duplicates
+       {1, 2, std::nullopt},
+       {std::nullopt, std::nullopt},
+       // 6th row: single set with non-null duplicates and nulls
+       {1, 2, 2, std::nullopt, std::nullopt},
+       // 7th row: single set with pure nulls
+       {std::nullopt, std::nullopt}});
+
+  std::vector<vector_size_t> sizes = {2, 3, 2, 3, 2, 1, 1};
+  // offsets array is cumulative sums starting from Offset 0
+  std::vector<vector_size_t> offsets(sizes.size()-1);
+  std::partial_sum(sizes.begin(), sizes.end(), offsets.begin(), std::plus<>());
+  offsets.emplace(offsets.begin(), 0);
+
+  // Create an array of array vector using above base vector
+  auto arrayOfArrays = createArrayVector(offsets, sizes, baseVector);
+
+  auto expected = makeNullableArrayVector<int64_t>(
+      {{2, 3},
+       {6},
+       {},
+       {std::nullopt, 2},
+       {std::nullopt},
+       {1, 2, 2, std::nullopt, std::nullopt},
+       {std::nullopt, std::nullopt}});
+
+  auto actual = evaluate<ArrayVector>(
+      "array_intersect(c0)", makeRowVector({arrayOfArrays}));
+
+  assertEqualVectors(expected, actual);
+}
+
+TEST_F(ArrayIntersectTest, arrayOfDoubleArrays) {
```

Comment:
I think it ought to be possible to templatize the test for both and int and double as we arent testing anything additional here.